proto/spire-next/api/server/agent/v1/agent.proto

```diff
@@ -0,0 +1,183 @@
+    // The caller is not authenticated.
+    rpc AttestAgent(stream AttestAgentRequest) returns (stream AttestAgentResponse);
+
+    // Renews the agent and returns a new X509-SVID using a bidirectional
+    // stream to facilitate two-phase commit of the renewal.
+    //
+    // The caller must present an active agent X509-SVID, i.e. the X509-SVID
+    // returned by the AttestAgent or the most recent RenewAgent call.
+    rpc RenewAgent(stream RenewAgentRequest) returns (stream RenewAgentResponse);
+
+    // Creates an agent join token. The token can be used with `join_token`
+    // attestation to join the trust domain.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc CreateJoinToken(CreateJoinTokenRequest) returns (spire.types.JoinToken);
+}
+
+message ListAgentsRequest {
+    message Filter {
+        // Filters agents to those matching the attestation type.
+        string by_attestation_type = 1;
+
+        // Filters agents to those satisfying the selector match.
+        spire.types.SelectorMatch by_selector_match = 2;
+
+        // Filters agents to those that are banned.
+        bool by_banned = 3;
+    }
+
+    // Filters the agents returned by the list operation.
+    Filter filter = 1;
+
+    // An output mask indicating which agent fields are set in the response.
+    spire.types.AgentMask output_mask = 2;
+
+    // The maximum number of items to return.
+    int32 page_size = 3;
+
+    // The next_page_token value returned from a previous List request, if any.
+    string page_token = 4;
+}
+
+message ListAgentsResponse {
+    // The agents.
+    repeated spire.types.Agent agents = 1;
+
+    // The page token for the next request. Empty if there are no more results.
+    string next_page_token = 2;
+}
+
+message GetAgentRequest {
+    // Required. The SPIFFE ID of the agent.
+    spire.types.SPIFFEID id = 1;
+
+    // An output mask indicating which agent fields are set in the response.
+    spire.types.AgentMask output_mask = 2;
+}
+
+message DeleteAgentRequest {
+    // Required. The SPIFFE ID of the agent.
+    spire.types.SPIFFEID id = 1;
+}
+
+message BanAgentRequest {
+    // Required. The SPIFFE ID of the agent.
+    spire.types.SPIFFEID id = 1;
+}
+
+message AttestAgentRequest {
+    message Params {
+        // Required. The attestation data.
+        spire.types.AttestationData data = 1;
+
+        // Required. The X509-SVID parameters.
+        AgentX509SVIDParams params = 2;
+    }
+
+    // Required. The data for the step in the attestation flow.
+    oneof step {
+        // Attestation parameters. These are only sent in the initial request.
+        Params params = 1;
+
+        // The response to a challenge issued by the attestor. Only sent in
+        // response to a challenge received by the issuer.
+        bytes challenge_response = 2;
+    }
+}
+
+message AttestAgentResponse {
+    message Result {
+        // The agent X509-SVID.
+        spire.types.X509SVID identity = 1;
```

Comment:
We name this as `svid` in other messages. Do you think that `identity` is a better choice here?
I'm just looking from a consistency point of view.